[BUG] test_window_aggs_for_negative_rows_partitioned failure in CI

**Describe the bug**

A CI build failed against Spark 3.3.4

```
[2024-01-12T17:29:50.254Z] FAILED ../../src/main/python/window_function_test.py::test_window_aggs_for_negative_rows_partitioned[-2 PRECEDING AND 4 FOLLOWING-[('a', RepeatSeq(Long)), ('b', Decimal(23,10)), ('c', Decimal(23,10))]-1000][DATAGEN_SEED=1705076302, INJECT_OOM, IGNORE_ORDER({'local': True}), APPROXIMATE_FLOAT] - AssertionError: GPU and CPU decimal values are different at [122, 'max_c_desc']
```

**Steps/Code to reproduce bug**


**Expected behavior**

**Environment details (please complete the following information)**

**Additional context**

--- a/integration_tests/src/main/python/window_function_test.py
+++ b/integration_tests/src/main/python/window_function_test.py
@@ -27,22 +27,22 @@ import warnings
 _grpkey_longs_with_no_nulls = [
     ('a', RepeatSeqGen(LongGen(nullable=False), length=20)),
     ('b', IntegerGen()),
-    ('c', IntegerGen())]
+    ('c', UniqueLongGen())]
 
 _grpkey_longs_with_nulls = [
     ('a', RepeatSeqGen(LongGen(nullable=(True, 10.0)), length=20)),
     ('b', IntegerGen()),
-    ('c', IntegerGen())]
+    ('c', UniqueLongGen())]
 
 _grpkey_longs_with_dates = [
     ('a', RepeatSeqGen(LongGen(), length=2048)),
     ('b', DateGen(nullable=False, start=date(year=2020, month=1, day=1), end=date(year=2020, month=12, day=31))),
-    ('c', IntegerGen())]
+    ('c', UniqueLongGen())]
 
 _grpkey_longs_with_nullable_dates = [
     ('a', RepeatSeqGen(LongGen(nullable=False), length=20)),
     ('b', DateGen(nullable=(True, 5.0), start=date(year=2020, month=1, day=1), end=date(year=2020, month=12, day=31))),
-    ('c', IntegerGen())]
+    ('c', UniqueLongGen())]
 
 _grpkey_longs_with_timestamps = [
     ('a', RepeatSeqGen(LongGen(), length=2048)),
@@ -57,17 +57,17 @@ _grpkey_longs_with_nullable_timestamps = [
 _grpkey_longs_with_decimals = [
     ('a', RepeatSeqGen(LongGen(nullable=False), length=20)),
     ('b', DecimalGen(precision=18, scale=3, nullable=False)),
-    ('c', DecimalGen(precision=18, scale=3))]
+    ('c', UniqueLongGen())]
 
 _grpkey_longs_with_nullable_decimals = [
     ('a', RepeatSeqGen(LongGen(nullable=(True, 10.0)), length=20)),
     ('b', DecimalGen(precision=18, scale=10, nullable=True)),
-    ('c', DecimalGen(precision=18, scale=10, nullable=True))]
+    ('c', UniqueLongGen())]
 
 _grpkey_longs_with_nullable_larger_decimals = [
     ('a', RepeatSeqGen(LongGen(nullable=(True, 10.0)), length=20)),
     ('b', DecimalGen(precision=23, scale=10, nullable=True)),
-    ('c', DecimalGen(precision=23, scale=10, nullable=True))]
+    ('c', UniqueLongGen())]
 
 _grpkey_longs_with_nullable_largest_decimals = [
     ('a', RepeatSeqGen(LongGen(nullable=(True, 10.0)), length=20)),

